File: Front-end/js/app/models/Produto.js
class Produto {

    constructor(marca, descricao, preco, quantidade, id) {
        this._marca = marca;
        this._descricao = descricao;
        this._preco = preco;
        this._quantidade = quantidade;
        this._id = id;
        Object.freeze(this);
    }

    get marca() {
        return this._marca;
    }

    get descricao() {
        return this._descricao;
    }

    get preco() {
        return this._preco;
    }

    get quantidade() {
        return this._quantidade;
    }

    get id() {
        return this._id;
    }

    toString() {
        return {
            marca: this._marca,
            descricao: this._descricao,
            preco: this._preco,
            quantidade: this._quantidade
        }
    }
}

File: Back-end/api/src/main/java/br/com/loja/api/model/Produto.java
package br.com.loja.api.model;

import java.math.BigDecimal;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "produtos")
public class Produto {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private String marca;
	private String descricao;
	private BigDecimal preco;
}
